torch/fx/experimental/GraphManipulation.py

```diff
@@ -53,3 +55,77 @@ def replace_target_nodes_with(
         else:
             val_map[node] = new_graph.node_copy(node, lambda n : val_map[n])
     fx_module.graph = new_graph
+
+def get_size_of_all_nodes(fx_module: GraphModule, args: List[torch.Tensor]) -> None:
+    """Given a fx graph module, update each node with its total size (weights + bias + output)
+    and its output_size(output). For a non-module node, the total size is the output size.
+    return total size"""
+    # Mark shape and dtype for each node (node.shape and node.dtype)
+    ShapeProp(fx_module).propagate(*args)
+    # Calculate the total size of the whole fx graph
+    total_size_of_graph = 0.0
+    for node in fx_module.graph.nodes:
+        if node.name == 'output':
+            break
+        node.size = get_size_of_node(fx_module, node)
+    return
+
+def get_size_of_node(fx_module: GraphModule, node: Node) -> Tuple[str, int]:
+    """Given a node, return its total size and its output size.
+       total_size = weights + bias + output_size
+    """
+    def product_of_list(size_list: List[int]):
+        product = 1
+        for i in size_list:
+            product *= i
+        return product
+
+    # Total num of elements (weights + bias + output)
+    total_num_of_elems = 0
+    # For a module, conside weights and bias
+    if node.op == 'call_module':
+        submodule_dict = dict(fx_module.named_modules())
+        if node.name in submodule_dict:
+            submodule = submodule_dict[node.name]
+        else:
+            tokens = node.name.split('_')
+            module_name = '_'.join(tokens[:-1])
+            submodule = submodule_dict[module_name]
+        parameters = submodule.named_parameters()
+        # Parameters are named tuples
+        for p in parameters:
+            if 'weight' in p or 'bias' in p:
+                total_num_of_elems += product_of_list(list(p[1].size()))
+    # Don't forget the output size
+    # node.shape is the shape of this node's output
+    assert hasattr(node, 'shape')
+    output_elem = product_of_list(list(node.shape))
+    total_num_of_elems += output_elem
+    # Num of bits for each dtype
+    dtype_to_bits = {
+        torch.float32 : 32,
+        torch.float : 32,
+        torch.float64: 64,
+        torch.double: 64,
+        torch.complex64: 64,
+        torch.cfloat: 64,
+        torch.complex128: 128,
+        torch.cdouble: 128,
+        torch.float16: 16,
+        torch.half: 16,
+        torch.bfloat16: 16,
+        torch.uint8: 8,
+        torch.int8: 8,
+        torch.int16: 16,
+        torch.short: 16,
+        torch.int32: 32,
+        torch.int: 32,
+        torch.int64: 64,
+        torch.long: 64,
+        torch.bool: 8
+    }
```

Comment:
One thing you can do instead of this table is use `element_size()` (https://pytorch.org/docs/stable/tensors.html#torch.Tensor.element_size) to retrieve the size of the dtype in bytes by creating an empty tensor with that dtype. Something like:
```
size_per_elem = torch.tensor([], dtype=node.dtype).element_size() * 8
```

I'm going to keep looking and see if there's a way to do this more directly without making a Tensor

Reply:
We could see about binding https://github.com/pytorch/pytorch/blob/14997f2125aa9d49edda94043c1ea9b8f2a692ef/c10/core/ScalarType.h#L242 into Python. Likely we could just expose it here: https://github.com/pytorch/pytorch/blob/14997f2125aa9d49edda94043c1ea9b8f2a692ef/torch/csrc/Dtype.cpp#L69

But if we do so, it should be in a separate PR

Reply:
I've also just been informed of `torch.finfo` and `torch.iinfo`, which have a `bits` attribute that you can retrieve: https://pytorch.org/docs/stable/type_info.html

Only downside is you have to determine if the `dtype` is a floating point or integral type